flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -212,21 +171,13 @@ public void start(
 
         resourceManagerId = Preconditions.checkNotNull(newResourceManagerId);
         mainThreadExecutor = Preconditions.checkNotNull(newMainThreadExecutor);
-        resourceAllocator = Preconditions.checkNotNull(newResourceAllocator);
         resourceEventListener = Preconditions.checkNotNull(newResourceEventListener);
         slotStatusSyncer.initialize(
                 taskManagerTracker, resourceTracker, resourceManagerId, mainThreadExecutor);
         blockedTaskManagerChecker = Preconditions.checkNotNull(newBlockedTaskManagerChecker);
+        taskManagerTracker.initialize(newResourceAllocator, mainThreadExecutor);
```

Comment:
```suggestion
        taskManagerTracker.initialize(Preconditions.checkNotNull(newResourceAllocator), mainThreadExecutor);       
```
We'd better do this check before pass parameter, otherwise we have to rely on all implementations.